packages/core-flows/src/region/steps/upsert-and-replace-region-payment-providers.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  ModuleRegistrationName,
+  Modules,
+  RemoteLink,
+} from "@medusajs/modules-sdk"
+import {
+  FilterableRegionProps,
+  IPaymentModuleService,
+  IRegionModuleService,
+  RemoteQueryFunction,
+} from "@medusajs/types"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+import {
+  arrayDifference,
+  ContainerRegistrationKeys,
+  MedusaError,
+  promiseAll,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+
+export interface UpsertAndReplaceRegionPaymentProvidersStepInput {
+  input:
+    | {
+        region_selector: FilterableRegionProps
+        payment_providers: string[]
+      }
+    | {
+        regions: { id: string }[]
+        payment_providers: string[]
+      }[]
+}
+
+type LinkItems = {
+  [Modules.REGION]: { region_id: string }
+  [Modules.PAYMENT]: { payment_provider_id: string }
+}[]
+
+async function validatePaymentProvidersExists(
+  paymentService: IPaymentModuleService,
+  paymentProviderIds: string[]
+) {
+  const paymentProviders = await paymentService.listPaymentProviders({
+    id: { $in: paymentProviderIds },
+    is_enabled: true,
+  })
+
+  const retrievedPaymentProviderIds = paymentProviders.map((p) => p.id)
+
+  const missingProviders = arrayDifference(
+    paymentProviderIds,
+    retrievedPaymentProviderIds
+  )
+
+  if (missingProviders.length) {
+    throw new MedusaError(
+      MedusaError.Types.NOT_FOUND,
+      `Payment providers with ids ${missingProviders.join(
+        ", "
+      )} not found or not enabled`
+    )
+  }
+}
+
+async function getCurrentRegionPaymentProvidersLinks(
+  regionIds: string[],
+  { remoteQuery }: { remoteQuery: RemoteQueryFunction }
+): Promise<
+  {
+    [Modules.REGION]: { region_id: string }
+    [Modules.PAYMENT]: { payment_provider_id: string }
+  }[]
+> {
+  const query = remoteQueryObjectFromString({
+    // TODO: use service: LINKS.RegionPaymentProvider, for that, the links.ts must be moved to the utils package, so that for link querying we can use link const instead of plain strings
```

Comment:
suggestion: Should we go ahead and do that now? Seems like a minor change, so I think it's OK to include it

Reply:
I think to not increase the PR size, I ll move the links and modules to the utils, I ll update the link usage but not the Modules enum to not impact all modules, I ve put a todo to say that we should now use the Modules that comes from the utils package instead of the one from the modules-sdk